src/MainWindow.vala

```diff
@@ -354,32 +356,77 @@ namespace Terminal {
-            style_popover_grid.margin = 12;
-            style_popover_grid.column_spacing = 6;
-            style_popover_grid.row_spacing = 12;
-            style_popover_grid.width_request = 200;
-            style_popover_grid.attach (font_size_grid, 0, 0, 3, 1);
-            style_popover_grid.attach (color_button_white, 0, 1, 1, 1);
-            style_popover_grid.attach (color_button_light, 1, 1, 1, 1);
-            style_popover_grid.attach (color_button_dark, 2, 1, 1, 1);
-            style_popover_grid.show_all ();
-
-            var style_popover = new Gtk.Popover (null);
-            style_popover.add (style_popover_grid);
-
-            var style_button = new Gtk.MenuButton ();
-            style_button.set_can_focus (false);
-            style_button.image = new Gtk.Image.from_icon_name ("open-menu-symbolic", Gtk.IconSize.SMALL_TOOLBAR);
-            style_button.popover = style_popover;
-            style_button.tooltip_text = _("Style");
-            style_button.valign = Gtk.Align.CENTER;
+            var color_grid = new Gtk.Grid ();
+            color_grid.column_homogeneous = true;
+            color_grid.margin_start = color_grid.margin_end = 12;
+            color_grid.margin_bottom = 6;
+
+            color_grid.add (color_button_white);
+            color_grid.add (color_button_light);
+            color_grid.add (color_button_dark);
+
+            var natural_copy_paste_label = new Gtk.Label (_("Natural Copy/Paste"));
+            natural_copy_paste_label.halign = Gtk.Align.START;
+            natural_copy_paste_label.hexpand = true;
+
+            var natural_copy_paste_switch = new Gtk.Switch ();
+            natural_copy_paste_switch.halign = Gtk.Align.END;
+
+            var natural_copy_paste_description = new Gtk.Label (_("<small>Shortcuts don’t require Shift, but may interfere with some interactive CLI apps</small>"));
+            natural_copy_paste_description.max_width_chars = 30;
+            natural_copy_paste_description.use_markup = true;
+            natural_copy_paste_description.wrap = true;
+            natural_copy_paste_description.xalign = 0;
+            natural_copy_paste_description.get_style_context ().add_class (Gtk.STYLE_CLASS_DIM_LABEL);
+
+            var natural_copy_paste_revealer = new Gtk.Revealer ();
+            natural_copy_paste_revealer.add (natural_copy_paste_description);
+            natural_copy_paste_revealer.reveal_child = Application.settings.get_boolean ("natural-copy-paste");
+
+            var natural_copy_paste_grid = new Gtk.Grid ();
+            natural_copy_paste_grid.margin_start = natural_copy_paste_grid.margin_end = 12;
+
+            natural_copy_paste_grid.attach (natural_copy_paste_label, 0, 0);
+            natural_copy_paste_grid.attach (natural_copy_paste_switch, 1, 0);
+            natural_copy_paste_grid.attach (natural_copy_paste_revealer, 0, 1, 2);
+
+            var natural_copy_paste_button = new Gtk.Button ();
```

Comment:
Instead of a regular button with style classes I think what would be better is using Gtk.ModelButton and then removing  its child like here: https://github.com/elementary/wingpanel-indicator-session/blob/master/src/Indicator.vala#L93

Right now there's some weird focus style stuff and that would fix it. You also wouldn't need to then add any margin to the grid